### ISSUE TITLE
update k3s.service for  RedHat  kernel 4.18.0-305.el8.x86_64

### ISSUE THREAD

**Issue by becash143** [k3s-io/k3s#4992]
`Job for k3s.service failed because the control process exited with error code.
See "systemctl status k3s.service" and "journalctl -xe" for details.`
 `kernel version: 4.18.0-305.el8.x86_64`
`k3s version v1.18.12+k3s1 (56cd3630)`

**Comment by becash143** [comment]
Changed k3s service parameters from ExecStartPre=/bin/sh -xc '! /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service' “ExecStartPre=/bin/sh -xc ' /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service'

**Comment by brandond** [comment]
That is not the correct fix. That ExecStartPre is in there to make sure that `nm-cloud-setup.service` IS NOT enabled. K3s will not work if this service is running - you must DISABLE it, instead of reversing the check in the k3s systemd unit.

See: https://rancher.com/docs/k3s/latest/en/advanced/#additional-preparation-for-red-hat-centos-enterprise-linux

**Comment by becash143** [comment]
Previously k3s was working. Now the coredns  is crashing. I am getting this issue metric-server
```
Error: unable to load configmap based request-header-client-ca-file: Get "https://10.43.0.1:443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication": dial tcp 10.43.0.1:443: i/o timeout
```
On ubuntu the installtion is Successful. On Redhat facing issues.
```
Feb 09 09:45:20 ip-172-31-82-9.ec2.internal k3s[3082]: , Header: map[Content-Type:[text/plain; charset=utf-8] X-Content-Type-Options:[nosniff]]
Feb 09 09:45:20 ip-172-31-82-9.ec2.internal k3s[3082]: I0209 09:45:20.158456    3082 controller.go:129] OpenAPI AggregationController: action for item v1beta1.metrics.k8s.io: Rate Limited Requeue.
Feb 09 09:45:21 ip-172-31-82-9.ec2.internal k3s[3082]: I0209 09:45:21.780678    3082 scope.go:110] "RemoveContainer" containerID="5fb30ccdcf3a626f11b651bd741870d6f83febfad27c1e1594aab71855af0125"
Feb 09 09:45:21 ip-172-31-82-9.ec2.internal k3s[3082]: E0209 09:45:21.780938    3082 pod_workers.go:836] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"metrics-server\" with CrashL>
Feb 09 09:45:29 ip-172-31-82-9.ec2.internal k3s[3082]: I0209 09:45:29.780110    3082 scope.go:110] "RemoveContainer" containerID="fbdaed699a5e784110ce4cc2d1b071950848c4200396503ae639c92a965eedaa"
Feb 09 09:45:29 ip-172-31-82-9.ec2.internal k3s[3082]: E0209 09:45:29.780393    3082 pod_workers.go:836] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"local-path-provisioner\" wit>
Feb 09 09:45:34 ip-172-31-82-9.ec2.internal k3s[3082]: I0209 09:45:34.780693    3082 scope.go:110] "RemoveContainer" containerID="5fb30ccdcf3a626f11b651bd741870d6f83febfad27c1e1594aab71855af0125"
Feb 09 09:45:34 ip-172-31-82-9.ec2.internal k3s[3082]: E0209 09:45:34.780943    3082 pod_workers.go:836] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"metrics-server\" with CrashL>
Feb 09 09:45:35 ip-172-31-82-9.ec2.internal k3s[3082]: E0209 09:45:35.717074    3082 resource_quota_controller.go:413] unable to retrieve the complete list of server APIs: metrics.k8s.io/v1beta1: the ser>
Feb 09 09:45:36 ip-172-31-82-9.ec2.internal k3s[3082]: W0209 09:45:36.239200    3082 garbagecollector.go:703] failed to discover some groups: map[metrics.k8s.io/v1beta1:the server is currently unable to >
```

**Comment by becash143** [comment]
Coredns logs
```
[WARNING] plugin/kubernetes: starting server with unsynced Kubernetes API
.:53
[INFO] plugin/reload: Running configuration MD5 = 442b35f70385f5c97f2491a0ce8a27f6
CoreDNS-1.8.4
linux/amd64, go1.16.4, 053c4d5
[INFO] plugin/ready: Still waiting on: "kubernetes"
[INFO] plugin/ready: Still waiting on: "kubernetes"
[INFO] plugin/ready: Still waiting on: "kubernetes"
[INFO] plugin/ready: Still waiting on: "kubernetes"
```

**Comment by caroline-suse-rancher** [comment]
Closing due to age - please reopen following our bug issue template if this bug can be reproduced on more current versions of k3s.

**Comment by brandond** [comment]
> The nm-cloud.service was not installed

it's nm-cloud-setup. not nm-cloud.

As discussed above, removing or inverting the logic of this test IS NOT A VALID FIX. The ExecStartPre only fails if the nm-cloud-setup.service is present and enabled, it DOES NOT prevent K3s from starting if this service is not installed, or if it is disabled.

If you see the service failing, confirm that the nm-cloud-setup service has been disabled, and that there are no other errors in the journald log for k3s.

**Comment by zjdaniels1985** [comment]
Sorry but you are not correct I don't have the package installed and k3s.servoxe will fail because it's not getting the status.

**Comment by zjdaniels1985** [comment]
Sorry but you are not correct I don't have the package installed and k3s.servoxe will fail because it's not getting the status. It's not supposed to fail if it's not installed but the error fails on that exact execution from the k3s.service file. Although I know removing the check is not a fix it's what I have to do until this issue is resolved.

**Comment by caroline-suse-rancher** [comment]
I'm going to lock this one now since it is closed - @zjdaniels1985 If you're still seeing this problem can you please open up a new issue following our bug report template (can be found here: https://github.com/k3s-io/k3s/issues/new/choose) with as much detail as possible so that we can try to reproduce it and identify what is going on?
